### PR TITLE
Add security module CLI option.

### PR DESCRIPTION
**Pull request by bgravenorst**
Signed-off-by: Byron Gravenorst <byron.gravenorst@consensys.net>

## PR description
Add security module CLI option.

### CODE REVIEW

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -2121,6 +2121,34 @@ rpc-ws-port="6174"
...
 [expose ports appropriately](../../HowTo/Find-and-Connect/Configuring-Ports.md).
 
+### security-module
+
+```bash tab="Syntax"
+--security-module=<NAME>
+```
+
+```bash tab="Command Line"
+--security-module=hsm_plugin
```
Comment (rain-on) [inline docs/Reference/CLI/CLI-Syntax.md:2131]: hsm_plugin? Maybe "crypto_provider". Its not necessarily using a hsm (could just be a local file)

File: docs/Reference/CLI/CLI-Syntax.md [hunk 2]
```diff
@@ -2121,6 +2121,34 @@ rpc-ws-port="6174"
...
+BESU_SECURITY_MODULE=hsm_plugin
+```
+
+```bash tab="Configuration File"
+security-module="hsm_plugin"
+```
+
+Name of the security module to use.
+
+Defaults to using the local private key file, and cannot be used with the
```
Comment (rain-on) [inline docs/Reference/CLI/CLI-Syntax.md:2144]: do you know what behaviour actually exists - i.e. do we show an error, or are we just suggesting people should use them together?

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Reference/CLI/CLI-Syntax.md:2131]: Based on the thread I tagged you in Byron maybe change `hsm_plugin` to `security_module` as the example and in the description add a _For example, a Hardware Security Module (HSM) or V3 filestore plugin_

File: docs/Reference/CLI/CLI-Syntax.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Reference/CLI/CLI-Syntax.md:2144]: @rain-on we can't test this in Pliny per the thread about this so don't know what the actual behaviour is.

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Syntax.md:2131]: Done.

**Pull request merged**